qiskit/extensions/standard/iden.py

```diff
@@ -15,61 +15,9 @@
 """
 Identity gate.
 """
-import numpy
-from qiskit.circuit import Gate
-from qiskit.circuit import QuantumCircuit
-from qiskit.util import deprecate_arguments
+import warnings
+# pylint: disable=unused-import
+from qiskit.extensions.standard.i import IGate, IdGate, i, iden
 
-
-class IdGate(Gate):
-    """Identity gate.
-
-    Identity gate corresponds to a single-qubit gate wait cycle,
-    and should not be optimized or unrolled (it is an opaque gate).
-    """
-
-    def __init__(self, label=None):
-        """Create new Identity gate."""
-        super().__init__("id", 1, [], label=label)
-
-    def inverse(self):
-        """Invert this gate."""
-        return IdGate()  # self-inverse
-
-    def to_matrix(self):
-        """Return a Numpy.array for the Id gate."""
-        return numpy.array([[1, 0],
-                            [0, 1]], dtype=complex)
-
-
-@deprecate_arguments({'q': 'qubit'})
-def iden(self, qubit, *, q=None):  # pylint: disable=unused-argument
-    """Apply Identity to to a specified qubit (qubit).
-
-    The Identity gate ensures that nothing is applied to a qubit for one unit
-    of gate time. It leaves the quantum states |0> and |1> unchanged.
-    The Identity gate should not be optimized or unrolled (it is an opaque gate).
-
-    Examples:
-
-        Circuit Representation:
-
-        .. jupyter-execute::
-
-            from qiskit import QuantumCircuit
-
-            circuit = QuantumCircuit(1)
-            circuit.iden(0)
-            circuit.draw()
-
-        Matrix Representation:
-
-        .. jupyter-execute::
-
-            from qiskit.extensions.standard.iden import IdGate
-            IdGate().to_matrix()
-    """
-    return self.append(IdGate(), [qubit], [])
-
-
-QuantumCircuit.iden = iden
+warnings.warn('This module is deprecated. The IdGate can now be found in i.py',
```

Comment:
Hmm.. this `warn` seems strange to me. I guess it's for the user doing `from qiskit.extensions import iden`... but why a user would do that?

Reply:
That's analogous to the deprecation of the controlled gate modules, e.g.: https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/extensions/standard/cx.py